File: RAI/Clase 5/python/Gato.py
class Gato:
    #No es necesario inicializar o declarar los atributos

    #Metodos
    #Método Constructor
    # la palabra reservada para indicar a si mismo -> self
    def __init__(self, name, sex, age, weight, color, texture):
        self.__name = name
        self.__sex = sex
        self.__age = age
        self.__weight = weight
        self.__color = color
        self.__texture = texture

    def eat(self):
        print(self.__name + " está comiendo...")

    def move(self):
        print(self.__name + " se está moviendo...")

    def meow(self):
        print(self.__name + " hace meoww!!!...")

    def purr(self):
        print(self.__name + " hace purrr!!!...")

    def hunt_mice(self):
        print(self.__name + " cazó a Mickey!...")

    def __str__(self):
        return "Mi nombre es " + self.__name + " y soy " + self.__sex + " y tengo " + str(self.__age) + " años."

File: RAI/Clase 5/java/Gato.java
//package RAI.Clase 5.java;

public class Gato {
    //Atributos
    private String name;
    private String sex;
    private int age;
    private int weight;
    private String color;
    private String texture;

    //Métodos
    //Método Constructor
    Gato(String name, String sex, int age, int weight, String color, String texture){
        //Palabra reservada para referirse a si mismo. -> this
        //this.name
        //asignación entre los valores externos(parametros) y los valores internos del objeto a través de this.
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.texture = texture;
    }

    void eat(){
        System.out.println(this.name + " está comiendo...");
    }

    void move(){
        System.out.println(this.name + " se está moviendo...");
    }

    void meow(){
        System.out.println(this.name + " hace meowww!!!...");
    }

    void purr(){
        System.out.println(this.name + " hace purrr!!!...");
    }

    void huntMice(){
        System.out.println(this.name + " cazó a Mickey!...");
    }

    public String toString(){
        //Devuelve los valores en formato String
        return "Mi nombre es " + this.name + " y soy " + this.sex + " y tengo " + this.age + " años.";
    }
}
